pinot-core/src/main/java/org/apache/pinot/core/operator/combine/MinMaxValueBasedSelectionOrderByCombineOperator.java

```diff
@@ -205,7 +205,7 @@ protected void processSegments() {
           ((AcquireReleaseColumnsSegmentOperator) operator).release();
         }
       }
-      PriorityQueue<Object[]> selectionResult = (PriorityQueue<Object[]>) resultsBlock.getRows();
+      PriorityQueue<Object[]> selectionResult = resultsBlock.getRowsAsPriorityQueue();
```

Comment:
We should be able to directly get the boundary value from the list. Converting it to PQ will eliminate the performance gain of keeping list

Reply:
I've uploaded a new version that retrieves the boundary in different ways depending on the data structure used in rows.